contrib/mpi-proxy-split/mpi-wrappers/mpi_collective_p2p.c

```diff
@@ -413,7 +413,7 @@ int MPI_Reduce(const void* sendbuf, void* recvbuf, int count,
   if (rank == root) {
     // Gather data into tmp_sendbuf at root
     int i;
-    char *tmp_sendbuf = (char *)malloc(count * extent);
+    char *tmp_sendbuf = (char *)malloc(count * extent * size);
```

Comment:
I believe this commit is incorrect.   `size` comes from `MPI_Comm_size` in `PROLOG_rank_size`.  The original `extent` should be correct.  I propose to remove this commit.

Would it be better if we rename `PROLOG_rank_size` to `PROLOG_Comm_rank_size`?

Reply:
Thanks for the correction.  It looks good.